plugins/inputs/consul/consul.go

```diff
@@ -0,0 +1,180 @@
+package consul
+
+import (
+	"net/http"
+	"reflect"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+var fieldMap = map[string]string{
+	"check_id":     "CheckID",
+	"check_name":   "Name",
+	"node":         "Node",
+	"service_id":   "ServiceID",
+	"service_name": "ServiceName",
+	"status":       "Status",
+	"notes":        "Notes",
+	"output":       "Output",
+}
+
+type HealthMetrics struct {
+	Fields []string
+	Tags   []string
+}
+
+type Consul struct {
+	Address    string
+	Scheme     string
+	Token      string
+	Username   string
+	Password   string
+	Datacentre string
+
+	// Path to CA file
+	SSLCA string `toml:"ssl_ca"`
+	// Path to host cert file
+	SSLCert string `toml:"ssl_cert"`
+	// Path to cert key file
+	SSLKey string `toml:"ssl_key"`
+	// Use SSL but skip chain & host verification
+	InsecureSkipVerify bool
+
+	// Health Check metrics specific config
+	HealthCheck HealthMetrics
+}
+
+var sampleConfig = `
+  ## Most of these values defaults to the one configured on a Consul's agent level.
+  ## Optional Consul server address (default: "localhost")
+  # address = "localhost"
+  ## Optional URI scheme for the Consul server (default: "http")
+  # scheme = "http"
+  ## Optional ACL token used in every request (default: "")
+  # token = ""
+  ## Optional username used for request HTTP Basic Authentication (default: "")
+  # username = ""
+  ## Optional password used for HTTP Basic Authentication (default: "")
+  # password = ""
+  ## Optional data centre to query the health checks from (default: "")
+  # datacentre = ""
+  # [inputs.consul.health_check]
```

Comment:
I'm not sure it's really necessary to make this user-configurable. Most plugins just hardcode reasonable defaults for these. For this plugin, it seems reasonable to just gather all of these metrics, since there are only 6 of them.

Same goes for the tags, unless there's a clear use-case where a user would want "node" to be a field instead of a tag.

Additionally, if users really don't want some of these metrics, they can use Telegraf's builtin filtering ability to get rid of them. (see https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#measurement-filtering)